Added tsd.json, updating tsconfig.json, wrote+added better-log.d.ts, …

SamuelMarks opened the pull request:
…updated test.ts::PathItem interface, added acorn.d.ts

RReverser commented on test.ts:
```diff
@@ -33,6 +34,7 @@ interface ProgramPathItem extends NodePathItem {
 }
 
 function diffAST(path: Array<PathItem>) {
+	console.log()
```
Why is this here?

SamuelMarks replied:
Whoops, that was just me testing code completion, will remove

RReverser commented:
If using tsd, please remove `typings` folder from the repo.

RReverser commented:
I guess we could even remove generated `*.js` files from `.git` but leave them in npm by splitting `.gitignore` and `.npmignore`.

SamuelMarks commented:
I've squashed my commits together. Not sure about your `.npmignore`+`.gitignore` idea. Your call.

RReverser commented:
Thanks. Looks like you removed source maps (by accident?), can you please return them back?

SamuelMarks commented:
Sure, don't recall removing them.

Hold on, what do you mean, just double checked my tree and both `test.js.map` & `index.js.map` are there. Are you referring to something else, like the `typings` folder? - Thought you asked me to remove that?

RReverser commented:
I'm referring to `compilerOptions.sourceMap` which caused changes like https://github.com/RReverser/typescript-estree/pull/2/files#diff-1dd241c4cd3fd1dd89c570cee98b79ddL91.

SamuelMarks commented:
@RReverser just sent through a commit, I think that's the change you requested?

RReverser commented:
Yes, thanks. But now... why did you remove Flow?
Please don't remove things without asking :)

SamuelMarks commented:
Oh, I replaced `estree` with `typings`, but looks like `flow` is in a different package to `estree`.

Just added that back in (with `tsd install flow --save`). New squashed commit coming your way momentarily!

SamuelMarks commented:
Actually hold off on merging this for now, the type definitions for `typescript.d.ts` are wrong, e.g. `ClassLikeDeclaration` isn't provided by the `typings` version. The `node_modules` version has `ClassLikeDeclaration`, but it isn't `export`ed…

EDIT: It is being exported by the [git version](https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts), so maybe I'll change the package.json version of typescript.

RReverser commented:
Yes, ESTree typings need periodical updates. I'll publish updated versions of them today or tomorrow.

SamuelMarks commented:
Sounds good.

How about the TypeScript side of things, which `d.ts` do you want to use? - The one from [`master`](https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts)?

PS: I did try changing from `"typescript": "^1.5.3"` to `"typescript": "^1.6.2"`, but that version still hadn't `export`ed `ClassLikeDeclaration`.

RReverser commented:
Ah I missed that you were talking about TypeScript not ESTree defs. Well... I just want whichever works if switching to tsd at all.

SamuelMarks commented:
Dropped an issue to their queue [https://github.com/Microsoft/TypeScript/issues/5186], in the meantime I'll send through a commit with their repo cloned into `cust_typings`.

SamuelMarks commented:
@RReverser just threw together another commit. I started from a straight clone of your repo, then begun the migration process.

Importantly it still doesn't compile properly, but I should be able to fix that in my next commit.

PS: You may be interested to know that Esprima is now written in TypeScript => https://twitter.com/basarat/status/664310408673521664
